x-pack/test/api_integration/apis/ml/job_audit_messages/get_job_audit_messages.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { omit } from 'lodash';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { getJobConfig } from './index';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  describe('get_job_audit_messages', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      for (const jobConfig of getJobConfig(2)) {
+        await ml.api.createAnomalyDetectionJob(jobConfig);
+      }
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should fetch all audit messages', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/job_audit_messages/messages`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body.messages.length).to.eql(2);
+      expect(omit(body.messages[0], 'timestamp')).to.eql({
+        job_id: 'test_get_job_audit_messages_2',
+        message: 'Job created',
+        level: 'info',
+        node_name: 'node-01',
+        job_type: 'anomaly_detector',
+      });
+      expect(omit(body.messages[1], 'timestamp')).to.eql({
+        job_id: 'test_get_job_audit_messages_1',
```

Comment:
Are we sure that the messages are always returned in that order?

Reply:
not sure, I can adjust the assertion code just in case

Reply:
Changed in https://github.com/elastic/kibana/pull/110793/commits/f7413cca404aad4a96d0d8d476dafe77787e9a0f